activerecord/test/cases/migration/references_foreign_key_test.rb

```diff
@@ -12,6 +12,7 @@ class ReferencesForeignKeyTest < ActiveRecord::TestCase
       teardown do
         @connection.drop_table "testings", if_exists: true
         @connection.drop_table "testing_parents", if_exists: true
+        @connection.drop_table "testing", if_exists: true
```

Comment:
can you move that `drop_table` into an ensure clause for the specific test. No need to run it for every other test if the table is only created in one of em.